content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,144 @@
+Network’s security is proportional to the value of stake committed.
+
+## Sharding and layer 2 solutions
+
+For solving this scaling issue, 'sharding' is introduced (e.g. in the case of [Ethereum 2.0](https://www.youtube.com/watch?v=ctzGr58_jeI&t=657s&ab_channel=Finematics)).
+Sharding is a way of spreading the computing and storage workload from a blockchain network, so that each node no longer has to process the entire network's transactional load.
+Each node only needs to maintain info corresponding to its specific shard or partition.
+For example, addresses starting with “0x00” will be part of shard 1 and addresses starting with “0x01” will be part of shard 2 etc.
+There is a mechanism in place for inter-shard communication so that the whole blockchain is still viewable from any node.
+All the shards will be connected to the “Beacon chain” which will have all the history with it and acts as an orchestrator of the whole network.
+The consensus between all the shards will be maintained by a "beacon chain".
+However, sharding has its own challenges.
+Corrupting nodes in a given shard may lead to permanent loss of data.
+One way to tackle this issue is by randomly assigning a node to a shard and randomly reassigning the node to another shard.
+Sharding has another problem that the composability aspect of smart contract compatible blockchain will no longer exist.
+For example, if a Decentralized Finance (DeFi) platform built on top of Ethereum relies on relies on or combines few other projects, then in a sharded system of Ethereum2.0, ideally they should be on the same shard which may not always be the case.
+If they are not on the same shard then they all needs to be ok with asynchronous updates which is [not ideal](https://www.coindesk.com/tech/2020/10/13/will-a-sharded-ethereum-be-flexible-enough-for-decentralized-finance/).
+
+## Avalanche
+
+The next level solution for scalability in terms of consensus algorithms is by Avalanche network which uses Snow Consensus with PoS as the underlying [sybil control mechanism](https://en.wikipedia.org/wiki/Sybil_attack).
+In Snow consensus, in short, a particular node randomly sub-samples from other nodes multiple times across multiple cycles and eventually arrives at a consensus.
+This is then recorded in a Directed Acyclic Graph (DAG) as opposed to a chain of blocks.
+Every node has its own view of the current consensus state; they do not progress in lock-step like they do for BFT algorithms.
+However, the local view of each node is eventually consistent with the view of all other nodes on the network.
+
+Snow consensus can easily scale to a higher level compared to PoW or PoS.
+Because the communicational complexity is much lower than BFT; we only need to communicate with random sub-samples of the network for each round.
```

Comment:
I'm not sure if I understood this correctly.. If I guessed right, perhaps this is better phrased:

```suggestion
Because the nodes only need to communicate with random sub-samples of the network for each round, the communicational complexity is much lower than BFT.
```